### PR TITLE
waf-137 add untangle branded 403 page

### PR DESCRIPTION
**Pull request by untanglejacob**

### CODE REVIEW

File: sync/nginx/error_conf_manager.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        file.flush()
+        file.close()
+
+    def write_403error_to_conf(self, settings_file, conf_file):
+        """write 403 error handling"""
+        #todo write custom 403 page to /403.html
+        conf_file.write("error_page 403 /403.html;\n")
+        conf_file.write("location = /403.html {\n")
+        conf_file.write("\troot /etc/error-pages;\n")
+        conf_file.write("\tsub_filter '$(ip)' '$remote_addr';\n")
```
Comment (jsommerville-untangle) [inline sync/nginx/error_conf_manager.py:58]: can we pass this directly as remote_ip or remote_addr? something to indicate the source of the IP in the variable we expose

File: sync/nginx/error_conf_manager.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        if "errorPages" not in settings_file.settings:
+            settings_file = self.create_error_page_settings(settings_file)
+
+    def sync_settings(self, settings_file, prefix, delete_list):
+        """sync settings"""
+        self.write_errors_to_conf(settings_file, prefix)
+
+    def create_error_page_settings(self, settings_file):
+        """create errorPages settings in settings_file. Empty object with custom html values"""
+        error_pages = {}
```
Comment (untanglejacob) [inline sync/nginx/error_conf_manager.py:30]: add a comment to denote what this map should be.
```
errorPages = {
   403: '/custom_403.html'
}
```

File: sync/nginx/error_conf_manager.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+"""This class is responsible for writing out the Error handling conf file"""
+from sync import registrar, Manager
+import os
+
+class ErrorsManager(Manager):
+    errors_conf="/etc/modsecurity.d/errors.conf"
```
Comment (jsommerville-untangle) [inline sync/nginx/error_conf_manager.py:6]: move this into /etc/nginx/ conf.d?

File: sync/nginx/error_conf_manager.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        if "errorPages" not in settings_file.settings:
+            settings_file = self.create_error_page_settings(settings_file)
+
+    def sync_settings(self, settings_file, prefix, delete_list):
+        """sync settings"""
+        self.write_errors_to_conf(settings_file, prefix)
+
+    def create_error_page_settings(self, settings_file):
+        """create errorPages settings in settings_file. Empty object with custom html paths"""
+        #errorPages = {
```
Comment (jsommerville-untangle) [inline sync/nginx/error_conf_manager.py:30]: add more comments around this commented code (this is placeholder for custom 403 pages)

File: sync/nginx/nginx_conf_manager.py [hunk 5]
```diff
@@ -268,10 +268,7 @@ def write_metrics_loc(self, file, nginx_locations):
 
     def write_error_pages(self, file, nginx_locations):
         """write_error_pages creates the http status code response pages"""
-        file.write("\terror_page 500 502 503 504  /50x.html;\n")
-        file.write("\tlocation = /50x.html {\n")
-        file.write("\t\troot /usr/share/nginx/html;\n")
-        file.write("\t}\n")
+        file.write('\tinclude /etc/modsecurity.d/errors.conf;\n')
```
Comment (jsommerville-untangle) [inline sync/nginx/nginx_conf_manager.py:271]: be sure this path is updated also with above comment

**Review by jsommerville-untangle — APPROVED** [review]

**Pull request merged**